### ISSUE TITLE
Cannot drop database when citus installed

### ISSUE THREAD

**Issue by onderkalaci** [citusdata/citus#1579]
The issue happens because the background daemon is already connected to the database, we should probably kill that process first.

```
-- connect to the main database
psql postgres
CREATE DATABASE test;

-- connect to the test database
psql test
CREATE EXTENSION citus;

-- issue a command to make sure that bg worker starts
\dx

-- connect back to the main database

postgres=# drop database test;
ERROR:  database "test" is being accessed by other users
DETAIL:  There is 1 other session using the database.
```

**Comment by onderkalaci** [comment]
This seems tricky to fix. 

The problem is that we want to terminate the background daemon only if the `DROP DATABASE` command succeeds.

We cannot terminate the daemon after the process utility since it'd definitely fail as described in the issue description due to the daemon's connection to the database (see `BackgroundWorkerInitializeConnectionByOid()`).

If we remove the db from the maintenance daemon before the standard process utility, we might terminate the daemon before actually the db is dropped (which may fail due to connections or other stuff).

Any ideas on how to do this without any hacks?

**Comment by marcocitus** [comment]
> The problem is that we want to terminate the background daemon only if the DROP DATABASE command succeeds.

Do we care that much? We could just let it terminate and remove it from the hash. If the `DROP DATABASE` rolls back and there are more commands, it will be restarted.

**Comment by sumedhpathak** [comment]
@onderkalaci Can we close this issue now?

**Comment by onderkalaci** [comment]
Yes, this is resolved with #1592. Closing the issue.
